### PR TITLE
Updated log levels

### PR DESCRIPTION
**Pull request by hrajagers**
This pull request addresses the C# version of the BMI interface. A related pull request exists for the C/C++ version.

### CODE REVIEW

File: src/BasicModelInterface/Logger.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
-﻿using System.Runtime.InteropServices;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace BasicModelInterface
 {
-    public enum Level { Off, Debug, Info, Error, Fatal };
+    public enum Level { LEVEL_ALL, LEVEL_DEBUG, LEVEL_INFO, LEVEL_WARNING, LEVEL_ERROR, LEVEL_FATAL, LEVEL_NONE };
```
Comment (gena) [inline src/BasicModelInterface/Logger.cs]: I'd better call them: public enum Level { All, Debug, Info, Warning, Error, Fatal, None }. This way it is still compatible with the C/C++ version, but also with the CSharp naming conventions: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions.

**Comment by hrajagers** [comment]
Point taken with respect to the use of underscores. Would public enum Level
{ LevelAll, LevelDebug, LevelInfo, LevelWarning, LevelError, LevelFatal,
LevelNone } not also be consistent with the naming conventions and be more
clear and explicit?

Best regards,

Bert

On 12 Sep 2017 11:28 am, "gena" <notifications@github.com> wrote:

*@gena* commented on this pull request.
------------------------------

In src/BasicModelInterface/Logger.cs
<https://github.com/openearth/bmi-csharp/pull/3#discussion_r138296383>:

>  using System.Text;

 namespace BasicModelInterface
 {
-    public enum Level { Off, Debug, Info, Error, Fatal };
+    public enum Level { LEVEL_ALL, LEVEL_DEBUG, LEVEL_INFO,
LEVEL_WARNING, LEVEL_ERROR, LEVEL_FATAL, LEVEL_NONE };

I'd better call them: public enum Level { All, Debug, Info, Warning, Error,
Fatal, None }. This way it is still compatible with the C/C++ version, but
also with the CSharp naming conventions: https://docs.microsoft.com/en-
us/dotnet/standard/design-guidelines/general-naming-conventions.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/openearth/bmi-csharp/pull/3#pullrequestreview-62070378>,
or mute the thread

**Comment by gena** [comment]
Not really, the problem is that in C# you have to use enum names when referring to enum values:

`Level l = Level.Info;`

But with the prefix, it will look awkward:

`Level l = Level.LevelInfo;`

For the C version, the use of prefix like LEVEL_INFO sounds perfectly logical, because of different syntax.

Some examples of C# enums:

```
enum Days { Saturday, Sunday, Monday, Tuesday, Wednesday, Thursday, Friday };
enum Colors { Red, Green, Blue, Yellow };
```

**Comment by hrajagers** [comment]
Thanks Gena for the additional information. Doubling the prefix indeed doesn't make sense then, I wasn't aware of that. I have updated the code to be consistent with your initial suggestions, so I hope that it's now suitable for merging.

**Pull request merged**